hazelcast/src/test/java/com/hazelcast/test/starter/HazelcastVersionLocator.java

```diff
@@ -16,105 +16,138 @@
-        files.put(Artifact.OS_TEST_JAR, locateArtifact(Artifact.OS_TEST_JAR, version));
-        if (Version.of(version).isGreaterOrEqual(Versions.V5_0)) {
-            files.put(Artifact.SQL_JAR, locateArtifact(Artifact.SQL_JAR, version));
+            if (!localCopy.exists()) {
+                downloadArtifact(version);
+            }
+            return localCopy;
         }
-        if (enterprise) {
-            files.put(Artifact.EE_JAR, locateArtifact(Artifact.EE_JAR, version));
-        }
-        return files;
-    }
 
-    private static File locateArtifact(Artifact artifact, String version) {
-        String path = format(artifact.path, version);
-        File localCopy = new File(LOCAL_M2_REPOSITORY_PREFIX + path);
-        if (!localCopy.exists()) {
-            downloadArtifact(artifact, version);
+        private static String getMvn() {
+            return OsHelper.isWindows() ? "mvn.cmd" : "mvn";
         }
-        return localCopy;
-    }
 
-    private static void downloadArtifact(Artifact artifact, String version) {
-        logWarningForArtifactDownload(artifact, version);
-        ProcessBuilder builder = new ProcessBuilder(buildMavenCommand(artifact, version).split(" "))
-                .inheritIO();
-        try {
-            Process process = builder.start();
-            boolean successful = process.waitFor(120, TimeUnit.SECONDS);
-            checkState(successful, "Maven dependency:get timed out");
-            checkState(process.exitValue() == 0, "Maven dependency:get failed");
-        } catch (InterruptedException | IOException e) {
-            throw new RuntimeException("Problem in invoking Maven dependency:get " + artifact + ":" + version , e);
+        private void downloadArtifact(final String version) {
+            // It's also possible to download this via Aether but I was unable to get this working
+            final ProcessBuilder builder = new ProcessBuilder(buildMavenCommand(version).toArray(String[]::new)).inheritIO();
+            try {
+                final Process process = builder.start();
+                final boolean successful = process.waitFor(2, TimeUnit.MINUTES);
+                checkState(successful, "Maven dependency:get timed out");
+                checkState(process.exitValue() == 0, "Maven dependency:get failed");
+            } catch (InterruptedException | IOException e) {
+                throw new RuntimeException("Problem in invoking Maven dependency:get " + toString() + ":" + version, e);
+            }
         }
-    }
 
-    private static void logWarningForArtifactDownload(Artifact artifact, String version) {
-        if (isOnJenkins()) {
-            return;
+        private Stream<String> buildMavenCommand(final String version) {
+            final Stream.Builder<String> builder = Stream.builder();
+
+            builder.add(getMvn());
+
+            builder.add("dependency:get");
+            builder.add("-DgroupId=" + GROUP_ID);
```

Comment:
add `builder.add("--quiet");` (and probably `--batch-mode` too) here, otherwise the forked JVM runner complains because the maven command output in stdout messes with surefire's std[in,out] streams-based inter-process communication (AFAIK):

```
[WARNING] Corrupted channel by directly writing to native stream in forked JVM 1. See FAQ web page and the dump file .../hazelcast/hazelcast/target/surefire-reports/2023-09-13T15-31-34_745-jvmRun1.dumpstream
```

Reply:
9c9b2fa6c115aeed9368910d03cc6b464eeebe71